tests/console/validate_multipath.pm

```diff
@@ -0,0 +1,172 @@
+
+    my $wwid = $test_data->{WWID};
+    my $dm_N = $test_data->{device_map}->{hd0};
+
+    record_info('topology', 'Verify multipath topology');
+    my $topology_output = script_output("multipath -l");
+
+    # Check infomation header
+    assert_matches(qr/$wwid $dm_N/, $topology_output,
+        "Topology information does not show header info with WWID dm-N");
+    # Check paths
+    while (my ($dev, $values) = each(%{$test_data->{topology}})) {
+        # Check policy
+        assert_matches(qr/prio=$values->{prio} status=$values->{status}/, $topology_output,
+            "Policy unexpected for multipath path to $dev");
+        # Check device
+        assert_matches(qr/$dev.*active.*running/, $topology_output,
+            "Device '$dev' should be active and running");
+    }
+}
+
+sub verify_paths_list {
+    my ($test_data) = shift;
+    my $wwid = $test_data->{WWID};
+
+    record_info('paths', 'Verify multipath paths');
+    while (my ($dev, $values) = each(%{$test_data->{paths_list}})) {
+        validate_script_output("multipath -d -v3 | grep ^$wwid",
+            qr/$wwid.*$dev.*$values->{vendor_product_revision}/);
+    }
+}
+
+sub verify_basic_multipath_configuration {
+    my ($test_data) = shift;
+
+    # Verify required packages are installed:
+    #   * device-mapper:   Device Mapper Tools
+    #   * multipath-tools: Tools to Manage Multipathed Devices with the device-mapper
+    #   * kpartx:          Manages partition tables on device-mapper devices
+    verify_packages_installed(['device-mapper', 'multipath-tools', 'kpartx']);
+
+    # Verify required kernel modules are loaded:
+    #   * dm_multipath: device-mapper multipath target
+    #   * dm_mod        device-mapper driver
+    #   * scsi_mod      SCSI core
+    verify_kernel_modules_loaded(['dm_multipath', 'dm_mod', 'scsi_mod']);
+
+    # Verify required service are running:
+    #   * multipathd.service: Device-Mapper Multipath Device Controller
+    verify_services_running(['multipathd']);
+
+    # A multipath device can be identified by its WWID, by a user-friendly name, or by an alias
+    # (Default) Use the WWIDs shown in the /dev/disk/by-id/ location.
+    verify_worldwide_id_used($test_data->{WWID});
+
+    # Verify internal device mapping
+    verify_devices_mapped($test_data);
+
+    # (Default) The multipath.conf file does not exist until you create and configure it.
+    verify_multipath_conf_file_not_exist();
+}
+
+sub verify_multipath {
+    my ($test_data) = shift;
+
+    # Verify the currently used multipathd configuration
+    verify_multipath_configuration($test_data);
+
+    # Verify the current multipath topology from information fetched in sysfs and the device mapper
+    verify_multipath_topology($test_data);
+
+    # Verify paths list
+    verify_paths_list($test_data);
+}
+
+sub run {
+    select_console 'root-console';
+    my $test_data = get_test_suite_data();
+
+    # Basic verification: checking packages, services, paths and files content
```

Comment:
I have doubt that your suggestion are correct in English gramma, but I'm not native. The first would mean something like '(some) contents of one file' and the second with the saxon genitive I'm not sure if it is right to use it because there is some rule that says that is better not to use it with objects in general.
I normally don't pluralize things too much as I could end up in a trap :) like this, but in this case I would go for:
'content of files'.